contracts/pool/concentrated/ConcentratedLiquidityPool.sol

```diff
@@ -289,56 +258,32 @@ contract ConcentratedLiquidityPool is IPool {
             );
         }
 
-        {
-            // Ensure no overflow happens when we cast to int128.
-            if (amount > uint128(type(int128).max)) revert Overflow();
-
-            uint256 amount0fees;
-            uint256 amount1fees;
-            (amount0fees, amount1fees, oldLiquidity) = _updatePosition(msg.sender, lower, upper, -int128(amount));
-
-            withdrawnAmounts = new TokenAmount[](2);
-            withdrawnAmounts[0] = TokenAmount({token: token0, amount: amount0});
-            withdrawnAmounts[1] = TokenAmount({token: token1, amount: amount1});
+        // Ensure no overflow happens when we cast from uint128 to int128.
+        if (amount > uint128(type(int128).max)) revert Overflow();
 
-            feesWithdrawn = new TokenAmount[](2);
-            feesWithdrawn[0] = TokenAmount({token: token0, amount: amount0fees});
-            feesWithdrawn[1] = TokenAmount({token: token1, amount: amount1fees});
+        (token0Fees, token1Fees) = _updatePosition(msg.sender, lower, upper, -int128(amount));
 
-            unchecked {
-                amount0 += amount0fees;
-                amount1 += amount1fees;
-            }
-        }
+        uint256 amount0;
+        uint256 amount1;
 
         unchecked {
+            amount0 = token0Amount + token0Fees;
+            amount1 = token1Amount + token1Fees;
             reserve0 -= uint128(amount0);
             reserve1 -= uint128(amount1);
         }
 
-        _transferBothTokens(recipient, amount0, amount1, unwrapBento);
+        _transferBothTokens(msg.sender, amount0, amount1, false);
```

Comment:
The `_transferBothTokens` function doesn't need to handle unwrapBento case anymore